### ISSUE TITLE
ReloadServlet ends up in endless loop

### ISSUE THREAD

**Issue by emilzse** [GeoWebCache/geowebcache#518]
When requesting _ReloadServlet_ we have experienced that it ends up in a endless loop and it logs the error in the logs until the storage is totally full.

The error comes from creating the request form:

```
public void doPost(Request req, Response resp) throws RestletException {
   Form form = req.getEntityAsForm(); // this never ends
```

The loop is in _com.noelios.restlet.util.FormReader#addParameters_ where it fails to parse a parameter and the while never ends.

```
// Let's read all form parameters
        while (readNext) {
            try {
                param = readNextParameter();

                if (param != null) {
                    // Add parsed parameter to the form
                    form.add(param);
                } else {
                    // Last parameter parsed
                    readNext = false;
                }
            } catch (IOException ioe) {
                getLogger().log(Level.WARNING,
                        "Unable to parse a form parameter. Skipping it.", ioe);
            }
        }
```

Looking at newer versions of this library (GWC use 1.0.8, newest is 1.1.10) they have moved the try around the while loop to break it if it ends up with exception.

http://www.atetric.com/atetric/javadoc/org.apache.servicemix.bundles/org.apache.servicemix.bundles.restlet/1.1.10_3/src-html/com/noelios/restlet/util/FormReader.html

One loop from the logs (this is repeated forever):

> 2017-05-17 05:21:30,120 INFO  (ajp-0.0.0.0-8009-7)[STDOUT] - com.noelios.restlet.util.FormReader::addParameters:318 - Unable to parse a form parameter. Skipping it.
> java.io.IOException: Socket read failed
>     at org.apache.coyote.ajp.AjpAprProcessor.read(AjpAprProcessor.java:1019)
>     at org.apache.coyote.ajp.AjpAprProcessor.readMessage(AjpAprProcessor.java:1141)
>     at org.apache.coyote.ajp.AjpAprProcessor.receive(AjpAprProcessor.java:1073)
>     at org.apache.coyote.ajp.AjpAprProcessor.refillReadBuffer(AjpAprProcessor.java:1112)
>     at org.apache.coyote.ajp.AjpAprProcessor.access$000(AjpAprProcessor.java:58)
>     at org.apache.coyote.ajp.AjpAprProcessor$SocketInputBuffer.doRead(AjpAprProcessor.java:1215)
>     at org.apache.coyote.Request.doRead(Request.java:428)
>     at org.apache.catalina.connector.InputBuffer.realReadBytes(InputBuffer.java:297)
>     at org.apache.tomcat.util.buf.ByteChunk.substract(ByteChunk.java:372)
>     at org.apache.catalina.connector.InputBuffer.readByte(InputBuffer.java:306)
>     at org.apache.catalina.connector.CoyoteInputStream.read(CoyoteInputStream.java:105)
>     at com.noelios.restlet.util.FormReader.readNextParameter(FormReader.java:231)
>     at com.noelios.restlet.util.FormReader.addParameters(FormReader.java:308)
>     at com.noelios.restlet.util.FormUtils.parsePost(FormUtils.java:92)
>     at com.noelios.restlet.Engine.parse(Engine.java:815)
>     at org.restlet.data.Form.<init>(Form.java:73)
>     at org.restlet.data.Form.<init>(Form.java:99)
>     at org.restlet.data.Message.getEntityAsForm(Message.java:150)
>     at org.geowebcache.rest.reload.ReloadRestlet.doPost(ReloadRestlet.java:54)
>     at org.geowebcache.rest.reload.ReloadRestlet.handle(ReloadRestlet.java:42)
>     at org.restlet.Filter.doHandle(Filter.java:105)
>     at org.restlet.Filter.handle(Filter.java:134)
>     at org.restlet.Router.handle(Router.java:444)
>     at com.noelios.restlet.ext.servlet.ServletConverter.service(ServletConverter.java:129)
>     at org.geowebcache.rest.RESTDispatcher.handleRequestInternal(RESTDispatcher.java:82)
>     at org.springframework.web.servlet.mvc.AbstractController.handleRequest(AbstractController.java:147)
>     at org.springframework.web.servlet.mvc.SimpleControllerHandlerAdapter.handle(SimpleControllerHandlerAdapter.java:50)
>     at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:959)
>     at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:893)
>     at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:968)
>     at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:870)
>     at javax.servlet.http.HttpServlet.service(HttpServlet.java:710)
>     at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:844)
>     at javax.servlet.http.HttpServlet.service(HttpServlet.java:803)
>     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)
>     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:316)
>     at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:126)
>     at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:90)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
>     at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
>     at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:122)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
>     at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
>     at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:169)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
>     at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:48)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
>     at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilterInternal(BasicAuthenticationFilter.java:158)
>     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
>     at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:64)
>     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
>     at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:91)
>     at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
>     at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:213)
>     at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:176)
>     at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346)
>     at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262)
>     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
>     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
>     at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:230)
>     at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:175)
>     at org.jboss.web.tomcat.security.SecurityAssociationValve.invoke(SecurityAssociationValve.java:179)
>     at org.jboss.web.tomcat.security.JaccContextValve.invoke(JaccContextValve.java:84)
>     at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
>     at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
>     at org.jboss.web.tomcat.service.jca.CachedConnectionValve.invoke(CachedConnectionValve.java:157)
>     at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
>     at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:262)
>     at org.apache.coyote.ajp.AjpAprProcessor.process(AjpAprProcessor.java:419)
>     at org.apache.coyote.ajp.AjpAprProtocol$AjpConnectionHandler.process(AjpAprProtocol.java:378)
>     at org.apache.tomcat.util.net.AprEndpoint$Worker.run(AprEndpoint.java:1508)
>     at java.lang.Thread.run(Thread.java:745)

There are many dependencies using the same _restlet.version_ in master _pom.xml_ so i am not sure if it is safe to upgrade this.

**Comment by smithkm** [comment]
I think this should be fixed in 1.12 with the switch from Rerstlet to Spring MVC, can you confirm that so we can close this?

**Comment by emilzse** [comment]
Yes you can close this
